### PR TITLE
Fixes #12405: move packages to host.

### PR DESCRIPTION
**Pull request by waldenraines**
Do not merge, just opening so others can take a look.  Will reopen against master after the host unification branch is merged.

http://projects.theforeman.org/issues/12405

### CODE REVIEW

**Comment by waldenraines** [comment]
[test]

File: app/controllers/katello/api/v2/host_packages_controller.rb [hunk 1]
```diff
@@ -43,15 +51,15 @@ def upgrade
       end
     end
 
-    api :PUT, "/systems/:system_id/packages/upgrade_all", N_("Update packages remotely"), :deprecated => true
-    param :system_id, :identifier, :required => true, :desc => N_("UUID of the content-host")
+    api :PUT, "/hosts/:host_id/packages/upgrade_all", N_("Update packages remotely"), :deprecated => true
+    param :host_id, :identifier, :required => true, :desc => N_("UUID of the content-host")
     def upgrade_all
       task     = async_task(::Actions::Katello::System::Package::Update, @system, [])
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/host_packages_controller.rb]: While updating each API, I'd like to see the dynflow actions updated as well.

File: test/controllers/api/v2/host_packages_controller_test.rb [hunk 2]
```diff
@@ -18,7 +19,12 @@ def setup
       login_user(User.find(users(:admin)))
       @request.env['HTTP_ACCEPT'] = 'application/json'
 
+      @host = hosts(:one)
       @system = katello_systems(:simple_server)
+
```
Comment (jlsherrill) [inline test/controllers/api/v2/host_packages_controller_test.rb]: nitpick probably don't need this newline

**Comment by jlsherrill** [comment]
If we are updating the old code (rather than adding new code), we need to make sure to update hammer while we are at it.

**Comment by waldenraines** [comment]
>If we are updating the old code (rather than adding new code), we need to make sure to update hammer while we are at it.

Good point, will address and reopen against master once #5630 goes in.

**Comment by waldenraines** [comment]
Closing.  Will reopen against master.

**Comment by waldenraines** [comment]
> Closing. Will reopen against master.

See #5656

**Pull request closed without merging**